app/models/user.rb

```diff
@@ -1,15 +1,7 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  # devise :database_authenticatable, :registerable,
-  #        :recoverable, :rememberable, :validatable, :confirmable
-  # has_secure_password
-
   has_many :reservations
   has_many :doctors
+  has_many :doctors, through: :reservations
```

Comment:
I think this is the same confusion for `has_many` and `has_many_through`, one choice is necessary instead of the two at the same time.

Reply:
Hey @moise-mulungu , I totally understand why you find it confusing. It may be confusing but it works very nicely. Let me show you the benefit of that relation in our case. 
In our case, we are creating a relation between a doctor and his patients through the reservation. With our association, we can do something like this : `doctor.users` to retrieve users or patients the doctor had reservations with. And in the other hand, we can do `doctor.user` to retrieve the user that created that doctor.